### PR TITLE
Feat/get txn hashes

### LINKED ISSUE #1365: Get transaction hashes by file metadata
A new REST API endpoint to return transaction hashes filtered by LookUpHash, Name or Contenthash. https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/dbs/event/writemarker.go#L29

### PR DESCRIPTION
**Pull request by lawRathod**
## Changes
Closes https://github.com/0chain/0chain/issues/1365

## Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber:
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

File: code/go/0chain.net/smartcontract/storagesc/handler.go [hunk 1]
```diff
@@ -1570,6 +1571,102 @@ func (srh *StorageRestHandler) getTransactionByFilter(w http.ResponseWriter, r *
...
+//
+// responses:
+//  200: stringArray
+//  400:
+//  500:
+func (srh *StorageRestHandler) getTransactionHashesByFilter(w http.ResponseWriter, r *http.Request) {
+	var (
+		lookUpHash   = r.URL.Query().Get("look_up_hash")
+		name         = r.URL.Query().Get("name")
+		contentHash  = r.URL.Query().Get("content_hash")
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/storagesc/handler.go]: use common API here after benchmark merge

File: code/go/0chain.net/smartcontract/storagesc/handler.go [hunk 2]
```diff
@@ -1570,6 +1571,102 @@ func (srh *StorageRestHandler) getTransactionByFilter(w http.ResponseWriter, r *
...
+//      in: query
+//      type: string
+//
+// responses:
+//  200: stringArray
+//  400:
+//  500:
+func (srh *StorageRestHandler) getTransactionHashesByFilter(w http.ResponseWriter, r *http.Request) {
+	var (
+		lookUpHash   = r.URL.Query().Get("look_up_hash")
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/storagesc/handler.go]: add index to WriteMarker

File: code/go/0chain.net/smartcontract/storagesc/handler.go [hunk 3]
```diff
@@ -1570,6 +1571,102 @@ func (srh *StorageRestHandler) getTransactionByFilter(w http.ResponseWriter, r *
...
+		name         = r.URL.Query().Get("name")
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/storagesc/handler.go]: Add index to WriteMarker

**Review by dabasov — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/smartcontract/storagesc/handler.go [hunk 4]
```diff
@@ -54,6 +54,7 @@ func GetEndpoints(rh rest.RestHandlerI) []rest.Endpoint {
 		rest.MakeEndpoint(storage+"/blobbers-by-geolocation", srh.getBlobbersByGeoLocation),
 		rest.MakeEndpoint(storage+"/transaction", srh.getTransactionByHash),
 		rest.MakeEndpoint(storage+"/transactions", srh.getTransactionByFilter),
+		rest.MakeEndpoint(storage+"/transactionHashes", srh.getTransactionHashesByFilter),
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/storagesc/handler.go]: let's use hyphen for URI, so `transaction-hashes`

File: code/go/0chain.net/smartcontract/storagesc/handler.go [hunk 5]
```diff
@@ -1570,6 +1571,102 @@ func (srh *StorageRestHandler) getTransactionByFilter(w http.ResponseWriter, r *
...
+// parameters:
+//    + name: look_up_hash
+//      description: restrict to transactions by the specific look up hash on write marker
+//      in: query
+//      type: string
+//    + name: name
+//      description: restrict to transactions by the specific file name on write marker
+//      in: query
+//      type: string
+//    + name: content_hash
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/storagesc/handler.go]: use hyphen `-` instead of underscore `_`, we will need to apply this for all URIs.

**Review by peterlimg — DISMISSED** [review]

**Comment by dabasov** [comment]
please add benchmark

File: code/go/0chain.net/smartcontract/storagesc/handler.go (hunk 2 above)
Comment (lawRathod) [reply code/go/0chain.net/smartcontract/storagesc/handler.go]: Done

File: code/go/0chain.net/smartcontract/storagesc/handler.go (hunk 1 above)
Comment (lawRathod) [reply code/go/0chain.net/smartcontract/storagesc/handler.go]: Done

File: code/go/0chain.net/smartcontract/storagesc/handler.go (hunk 3 above)
Comment (lawRathod) [reply code/go/0chain.net/smartcontract/storagesc/handler.go]: Done

File: code/go/0chain.net/smartcontract/dbs/event/writemarker.go [hunk 6]
```diff
@@ -128,3 +128,21 @@ func (wm *WriteMarker) exists(edb *EventDb) (bool, error) {
...
+	var wm []interface{}
+
+	edbRef := edb.Store.Get()
+	if len(selectString) > 0 {
+		edbRef = edbRef.Select(selectString)
+	}
+
+	res := edbRef.
+		Model(WriteMarker{}).
+		Offset(limit.Offset).
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/dbs/event/writemarker.go:142]: add ordering

File: code/go/0chain.net/smartcontract/storagesc/benchmark_rest_tests.go [hunk 7]
```diff
@@ -79,6 +79,17 @@ func BenchmarkRestTests(
 				},
 				Endpoint: srh.getTransactionByFilter,
 			},
+			{
+				FuncName: "transactions",
+				Params: map[string]string{
+					"offset":       "",
+					"limit":        "",
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/storagesc/benchmark_rest_tests.go]: leave them blank if not used

File: code/go/0chain.net/smartcontract/storagesc/handler.go [hunk 8]
```diff
@@ -1666,6 +1667,86 @@ func (srh *StorageRestHandler) getTransactionByFilter(w http.ResponseWriter, r *
...
+//      type: string
+//    + name: content-hash
+//      description: restrict to transactions by the specific content hash on write marker
+//      in: query
+//      type: string
+//    + name: offset
+//      description: offset
+//      in: query
+//      type: string
+//    + name: limit
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/storagesc/handler.go:1690]: add ordering param

**Review by dabasov — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/smartcontract/storagesc/handler.go (hunk 8 above)
Comment (lawRathod) [reply code/go/0chain.net/smartcontract/storagesc/handler.go:1690]: Done

File: code/go/0chain.net/smartcontract/storagesc/benchmark_rest_tests.go (hunk 7 above)
Comment (lawRathod) [reply code/go/0chain.net/smartcontract/storagesc/benchmark_rest_tests.go]: done

File: code/go/0chain.net/smartcontract/dbs/event/writemarker.go (hunk 6 above)
Comment (lawRathod) [reply code/go/0chain.net/smartcontract/dbs/event/writemarker.go:142]: done

**Review by dabasov — APPROVED** [review]
lgtm

**Pull request merged**